Azure.ResourceManager.Core.Tests/ResourceTypeTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Core.Tests
+{
+    public class ResourceTypeTests
+    {
+        const string resourcePrefix = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/";
+
+        [TestCase(0, "Microsoft.ClassicStorage/storageAccounts/account1",
```

Comment:
add a test to compare to null

Reply:
Since parameter null will throw ArgumentOutOfRangeException exception, I add object null compare here.